deployment/terraform/modules/docker_host/main.tf

```diff
@@ -0,0 +1,60 @@
+resource "aws_security_group" "allow_docker" {
+  name        = "${var.security_group_name}"
+  description = "Allow connection to docker engine from within VPC"
+  vpc_id      = "${var.vpc_id}"
+
+  # Explicitly set rule for port 2376 since this is the conventional port used
+  # for interacting with docker over tcp
+  ingress {
+    from_port   = 2376
+    to_port     = 2376
+    protocol    = "tcp"
+    cidr_blocks = ["${var.vpc_cidr_block}"]
+  }
+
+  # Allow access on IANA User ports:
+  # https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
+  ingress {
+    from_port   = 1024
+    to_port     = 49151
+    protocol    = "tcp"
+    cidr_blocks = ["${var.vpc_cidr_block}"]
+  }
+
+  egress {
+    from_port   = 0             # Implicit with AWS, but Terraform requires that it be explicit:
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags {
+    Name = "${var.security_group_name}"
+  }
+}
+
+resource "aws_instance" "docker_host" {
+  ami                    = "ami-2757f631"
+  subnet_id              = "${var.private_subnet_id}"
+  instance_type          = "t2.medium"
```

Comment:
The single cpu t2 instance types had poor performance while testing

Reply:
What were the testing conditions?

Reply:
I wanted to at the very least hit the max running container limit of 10 while having at least two instances of each of the tools to be deployed in production running. While trying that with `t2.micro` I was unable to launch more than 3 concurrently. I then bumped up to `t2.medium` and had success. I could always try with `t2.small`

Reply:
That would be useful. Over-provisioning becomes expensive very quickly: t2.small is twice the cost of t2.micro and t2.medium is twice the cost of t2.small. Scaling horizontally instead of vertically is key.